File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def solution(video_len, pos, op_start, op_end, commands):
    answer = ''
    dic = {"next": 10, "prev": -10}
    
    # 전체 비디오 길이 초 단위로 변경
    end_min, end_sec = video_len.split(":")
    end_total = int(end_min) * 60 + int(end_sec)
    
    # op_start, op_end 초 단위로 변경
    op_start_min, op_start_sec = op_start.split(":")
    op_end_min, op_end_sec = op_end.split(":")
    
    op_start_total = int(op_start_min) * 60 + int(op_start_sec)
    op_end_total = int(op_end_min) * 60 + int(op_end_sec)
    
    # pos 초 단위로 변경
    pos_min, pos_sec = pos.split(":")
    pos_total = int(pos_min) * 60 + int(pos_sec)
    
    if op_start_total <= pos_total < op_end_total:
            pos_total = op_end_total
            
    for i in commands:
        pos_total += dic[i]
        pos_total = max(0, pos_total) # 0보다 작아지면 0이 되게 함
        pos_total = min(end_total, pos_total) # 비디오 길이보다 길어지면 비디오 길이가 되게 함
        
        if op_start_total <= pos_total < op_end_total:
            pos_total = op_end_total
        
    if pos_total // 60 < 10:
        answer += "0" + str(pos_total // 60)
    else:
        answer += str(pos_total // 60)
    
    answer += ":"
    
    if pos_total % 60 < 10:
        answer += "0" + str(pos_total % 60)
    else:
        answer += str(pos_total % 60)
    
    # answer += str(pos_total // 60)
    # answer += ":"
    # answer += str(pos_total % 60)
    
    return answer

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
import java.util.HashMap;

class Solution {
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        String answer = "";
        HashMap<String, Integer> dic = new HashMap<>();
        dic.put("next", 10);
        dic.put("prev", -10);

        String[] endParts = video_len.split(":");
        int endTotal = Integer.parseInt(endParts[0]) * 60 + Integer.parseInt(endParts[1]);

        String[] opStartParts = op_start.split(":");
        String[] opEndParts = op_end.split(":");
        int opStartTotal = Integer.parseInt(opStartParts[0]) * 60 + Integer.parseInt(opStartParts[1]);
        int opEndTotal = Integer.parseInt(opEndParts[0]) * 60 + Integer.parseInt(opEndParts[1]);
        
        String[] posParts = pos.split(":");
        int posTotal = Integer.parseInt(posParts[0]) * 60 + Integer.parseInt(posParts[1]);

        
        if (opStartTotal <= posTotal && posTotal < opEndTotal) {
            posTotal = opEndTotal;
        }

        for (String command : commands) {
            posTotal += dic.get(command);
            posTotal = Math.max(0, posTotal); 
            posTotal = Math.min(endTotal, posTotal);

            if (opStartTotal <= posTotal && posTotal < opEndTotal) {
                posTotal = opEndTotal;
            }
        }

        if (posTotal / 60 < 10) {
            answer += "0" + (posTotal / 60);
        } else {
            answer += (posTotal / 60);
        }

        answer += ":";

        if (posTotal % 60 < 10) {
            answer += "0" + (posTotal % 60);
        } else {
            answer += (posTotal % 60);
        }

        return answer;
    }
}